sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala

```diff
@@ -68,6 +70,9 @@ class InMemoryFileIndex(
   refresh0()
 
   override def partitionSpec(): PartitionSpec = {
+    if (userSpecifiedPartitionSpec.isDefined) {
+      return userSpecifiedPartitionSpec.get
```

Comment:
how about `cachedPartitionSpec = userSpecifiedPartitionSpec.get`? then the code flow is more consistent and we will log the partition spec as well.

Reply:
yea, updated.